.github/actions/url-download-action/action.yaml

```diff
@@ -0,0 +1,72 @@
+---
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: 2025 The Linux Foundation
+
+# url-download-action
+name: "🌍 Download Content from URL"
+description: "Uses WGET to download the content at a given URL"
+
+inputs:
+  # Mandatory
+  URL:
+    description: "URL to download"
+    required: true
+  # Optional
+  EXIT_ON_FAIL:
+    description: "Exit with error if download fails"
+    required: false
+    default: true
+  NO_CLOBBER:
+    description: "Will not overwrite existing local file"
+    required: false
+    default: false
+  PATH:
+    description: "Saves file to given path"
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: "Verify WGET is available"
+      shell: bash
+      run: |
+        # Verify WGET is available
+        WGET_CMD=$(which wget)
+        if [ ! -x $WGET_CMD ]; then
+          echo "Unable to find/install WGET command ❌"
+          echo "Check instance type and OS environment"
+          exit 1
+        fi
+
+    - name: "Download content at URL"
+      shell: bash
+      # yamllint disable rule:line-length
+      run: |
+        # Download content at URL
+
+        # Process inputs, set variable/parameters
+        URL="${{ inputs.url }}"
+        if [ "${{ inputs.no_clobber }}" ]; then
```

Comment:
The input var is being treated as a boolean, but bash evaluates everything as strings. With a declared default of false then the string is "false" and the evaluation will always pass setting no_clobber because the string evaluates to true.

This should be changed to something like:

`if [ "f${{ inputs.no_clobber }}" == "ftrue" ]; then`

The reason for the `f` in both of those strings is to forcibly validate that the string is properly being evaluated. `f` is the a fairly common forcing string prefix for these sorts of tests